build.ps1

```diff
@@ -868,11 +870,21 @@ $items.GetEnumerator() | %{
 	$item.Name = $name
 	$item.ArchiveFile = $archiveFile
 	$item.PatchDirectory = $(New-Object System.IO.DirectoryInfo $patchDirectory)
-	$item.BuildDirectory = $(New-Object System.IO.DirectoryInfo "$workingDirectory\$($archiveFile.BaseName)")
-	$item.BuildArchiveFile = $(New-Object System.IO.FileInfo "$workingDirectory\$($archiveFile.BaseName)-$filenameArch$($archiveFile.Extension)")
 	$item.Dependencies = @($item.Dependencies | %{ $items[$_] })
 	$item.Dependents = @()
 	$item.State = ''
+
+	switch ([System.IO.Path]::GetExtension($item.ArchiveFile)) {
+		'.xz' {
```

Comment:
As I mentioned in my comment on the thread, I'd rather this check explicitly for `'.tar.xz'` instead, so you can't use Path.GetExtension nor a switch block for this.